pkg/registries/set_impl.go

```diff
@@ -53,19 +53,37 @@ func (e *setImpl) GetRegistryMetadataByImage(image *storage.Image) *types.Config
 	e.lock.RLock()
 	defer e.lock.RUnlock()
 
+	reg := e.getRegistryByImageNoLock(image)
+	if reg != nil {
+		return reg.Config()
+	}
+
+	return nil
+}
+
+// GetRegistryByImage returns the registry that contains the input image.
+func (e *setImpl) GetRegistryByImage(image *storage.Image) types.Registry {
+	e.lock.RLock()
+	defer e.lock.RUnlock()
+
+	return e.getRegistryByImageNoLock(image)
```

Comment:
Not related to your PR, but this function can refactored to reduce the critical section. Currently, sorting and matching are under lock which does not seem necessary.